src/main/java/org/wso2/carbon/inbound/salesforce/poll/EmpConnector.java

```diff
@@ -0,0 +1,394 @@
+        try {
+            httpClient.start();
+        } catch (Exception e) {
+            LOG.error("Unable to start HTTP transport[{}]", parameters.endpoint(), e);
+            running.set(false);
+            future.complete(false);
+            return future;
+        }
+        String bearerToken = bearerToken();
+        LongPollingTransport httpTransport = new LongPollingTransport(parameters.longPollingOptions(), httpClient) {
+            @Override
+            protected void customize(Request request) {
+
+                request.header(AUTHORIZATION, bearerToken);
+            }
+        };
+        client = new BayeuxClient(parameters.endpoint().toExternalForm(), httpTransport);
+        client.addExtension(new ReplayExtension(replay));
+        addListeners(client);
+        client.handshake((c, m) -> {
+            if (!m.isSuccessful()) {
+                Object error = m.get(ERROR);
+                if (error == null) {
+                    error = m.get(FAILURE);
+                }
+                future.completeExceptionally(
+                        new ConnectException(String.format("Cannot connect [%s] : %s", parameters.endpoint(), error)));
+                running.set(false);
+            } else {
+                subscriptions.forEach(SubscriptionImpl::subscribe);
+                future.complete(true);
+            }
+        });
+        return future;
+    }
+
+    private void addListeners(BayeuxClient client) {
+
+        for (MessageListenerInfo info : listenerInfos) {
+            client.getChannel(info.getChannelName()).addListener(info.getMessageListener());
+        }
+    }
+
+    private String bearerToken() {
+
+        String bearerToken;
+        if (bearerTokenProvider != null) {
+            bearerToken = bearerTokenProvider.apply(reauthenticate.get());
+            reauthenticate.compareAndSet(true, false);
+        } else {
+            bearerToken = parameters.bearerToken();
+        }
+        return bearerToken;
+    }
+
+    private void reconnect() {
+
+        if (running.compareAndSet(false, true)) {
+            connect();
+        } else {
+            LOG.error("The current value of running is not as we expect, this means our reconnection may not happen");
+        }
+    }
+
+    /**
+     * Listens to /meta/connect channel messages and handles 401 errors, where client needs
+     * to reauthenticate.
+     */
+    private class AuthFailureListener implements ClientSessionChannel.MessageListener {
+
+        private static final String ERROR_401 = "401";
+        private static final String ERROR_403 = "403";
+
+        @Override
+        public void onMessage(ClientSessionChannel channel, Message message) {
+
+            if (!message.isSuccessful()) {
+                if (isError(message, ERROR_401) || isError(message, ERROR_403)) {
+                    reauthenticate.set(true);
+                    stop();
+                    reconnect();

```

Comment:
As per the meaning of reconnecting, shall we call the stop() inside the reconnect() method?

Reply:
fixed